### PR TITLE
feat(docs): adding 'behaviors' section to menu

### PR DESCRIPTION
**Pull request by kolaps33**
# Accessibility description 

Changes done in this PR:
- adding section 'Behaviors' into the menu under the 'Components' section 
- in section 'Behaviors' all existing behaviors are listed
- inside the particular Behavior all different behaviors for the component are listed
- particular description for behavior will be stored in behavior file with tag '@description' - description it will be added in next PR

Please see screenshot with changes:
image

### CODE REVIEW

File: docs/src/examples/components/Menu/Types/MenuExample.shorthand.tsx [hunk 1]
```diff
@@ -2,9 +2,27 @@ import React from 'react'
 import { Menu } from '@stardust-ui/react'
 
 const items = [
-  { key: 'editorials', content: 'Editorials' },
-  { key: 'review', content: 'Reviews' },
-  { key: 'events', content: 'Upcoming Events' },
+  {
```
Comment (miroslavstastny) [inline docs/src/examples/components/Menu/Types/MenuExample.shorthand.tsx]: Please remove this debug changes

File: test/specs/components/Image/Image-test.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import * as React from 'react'
```
Comment (miroslavstastny) [inline test/specs/components/Image/Image-test.tsx]: The changed tests are not part of 'Accessibility documentation', these should be a separate PR.

**Review by miroslavstastny — CHANGES_REQUESTED** [review]
Please make sure this PR contains as few changes as possible. It seems that is contains a lot of changes not directly related to it.

File: test/specs/components/Image/Image-test.tsx (hunk 2 above)
Comment (jurokapsiar) [reply test/specs/components/Image/Image-test.tsx]: this actually is a separate PR (https://github.com/stardust-ui/react/pull/74/files) - once that gets merged, the change should disappear from this PR.

File: docs/src/components/Sidebar/Sidebar.tsx [hunk 3]
```diff
@@ -12,9 +12,12 @@ import { getComponentPathname, typeOrder, repoURL } from 'docs/src/utils'
 
 const pkg = require('../../../../package.json')
 const componentMenu = require('docs/src/componentMenu')
+const behaviorMenuItems = require('docs/src/componentMenuBehaviors')
 
 const selectedItemLabelStyle: any = { color: '#35bdb2', float: 'right' }
 const selectedItemLabel = <span style={selectedItemLabelStyle}>Press Enter</span>
+// assume all behaviors are the same type
+const behaviorType = behaviorMenuItems[0].type
```
Comment (smykhailov) [inline docs/src/components/Sidebar/Sidebar.tsx]: this is a dangerous assumption, the types might be different, or if all of them are the same, no need to introduce the type at all and this construction can also cause `undefined` error

File: docs/src/routes.tsx [hunk 4]
```diff
@@ -4,16 +4,21 @@ import { BrowserRouter, Redirect, Route, Switch } from 'react-router-dom'
 import ExternalExampleLayout from './components/ExternalExampleLayout'
 import DocsLayout from './components/DocsLayout'
 import DocsRoot from './components/DocsRoot'
+import DocsBehaviorRoot from './components/DocsBehaviorRoot'
 
 import Introduction from './views/Introduction'
 import PageNotFound from './views/PageNotFound'
 
+const behaviorMenuItems = require('docs/src/componentMenuBehaviors')
+const behaviorType = behaviorMenuItems[0].type
```
Comment (smykhailov) [inline docs/src/routes.tsx]: The same as comment above

File: docs/src/components/DocsBehaviorRoot.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
+import * as _ from 'lodash'
+import PropTypes from 'prop-types'
+import * as React from 'react'
+const behaviorMenuItems = require('docs/src/componentMenuBehaviors')
+import { Divider, Form, Grid, Menu, Segment, Visibility, SemanticCOLORS } from 'semantic-ui-react'
```
Comment (hughreeling) [inline docs/src/components/DocsBehaviorRoot.tsx]: Do we really need all these?  Don't we have stardust versions?

File: src/lib/accessibility/Behaviors/Icon/IconBehavior.ts [hunk 6]
```diff
@@ -1,5 +1,11 @@
 import { Accessibility } from '../../interfaces'
 
+/**
+ * @description
+ * The IconBehavior add attribute "aria-hidden='true'" on icon.
```
Comment (sophieH29) [inline src/lib/accessibility/Behaviors/Icon/IconBehavior.ts]: typo ```adds```, and I assume ```to icon```

File: src/lib/accessibility/Behaviors/Button/ToggleButtonBehavior.ts [hunk 7]
```diff
@@ -1,5 +1,13 @@
 import { Accessibility } from '../../interfaces'
 
+/**
+ * @description
+ * The ToggleButtonBehavior adds:
+ * - role='button' if element type is other than 'button'. This allows the screen readers handle component as button.
+ * - attribute 'aria-pressed' based on the active property. Based on this, screen readers can recognized state of button, if it is pressed or not.
```
Comment (sophieH29) [inline src/lib/accessibility/Behaviors/Button/ToggleButtonBehavior.ts]: typo ```recognize```

File: src/lib/accessibility/Behaviors/Image/ImageBehavior.ts [hunk 8]
```diff
@@ -1,5 +1,11 @@
 import { Accessibility } from '../../interfaces'
 
+/**
+ * @description
+ * The ImageBehavior add attribute "aria-hidden='true'"  on img tag, if there is no 'alt' attribute provided.
```
Comment (sophieH29) [inline src/lib/accessibility/Behaviors/Image/ImageBehavior.ts]: typo ```adds```, ```to  tag```

File: src/lib/accessibility/Behaviors/List/BasicListItemBehavior.ts [hunk 9]
```diff
@@ -1,5 +1,11 @@
 import { IAccessibilityDefinition } from '../../interfaces'
 
+/**
+ * @description
+ * The BasicListItemBehavior adds role='listitem'.
+ * The 'listitem' role is used to identify an element that a single item in a list.
```
Comment (sophieH29) [inline src/lib/accessibility/Behaviors/List/BasicListItemBehavior.ts]: typo 'that ```is``` a'

File: src/lib/accessibility/Behaviors/List/ListBehavior.ts [hunk 10]
```diff
@@ -2,6 +2,11 @@ import { Accessibility } from '../../interfaces'
 import SelectableListBehavior from './SelectableListBehavior'
 import BasicListBehavior from './BasicListBehavior'
 
+/**
+ * @description
+ * The ListBehavior adds beharior BasicListBehavior or SelectableListBehavior based on "selection" property.
```
Comment (sophieH29) [inline src/lib/accessibility/Behaviors/List/ListBehavior.ts]: I would say ```The ListBehavior defines a behavior based on "selection" property.```

or 

```The ListBehavior defines a behavior (BasicListBehavior or SelectableListBehavior) based on "selection" property.```

File: src/lib/accessibility/Behaviors/List/ListItemBehavior.ts [hunk 11]
```diff
@@ -2,6 +2,11 @@ import SelectableListItemBehavior from './SelectableListItemBehavior'
 import BasicListItemBehavior from './BasicListItemBehavior'
 import { IAccessibilityDefinition } from '../../interfaces'
 
+/**
+ * @description
+ * The ListItemBehavior adds beharior SelectableListItemBehavior or BasicListItemBehavior based on "selection" property.
```
Comment (sophieH29) [inline src/lib/accessibility/Behaviors/List/ListItemBehavior.ts]: same as above

File: src/lib/accessibility/Behaviors/List/SelectableListItemBehavior.ts [hunk 12]
```diff
@@ -1,5 +1,12 @@
 import { IAccessibilityDefinition } from '../../interfaces'
 
+/**
+ * @description
+ * The SelectableListItemBehavior adds :
+ * - role='option', this role is used for a selectable item in a list
+ * - attribute 'aria-selected' based on the active property. Based on this, screen readers can recognized state of item in list, if it is selected or not.
```
Comment (sophieH29) [inline src/lib/accessibility/Behaviors/List/SelectableListItemBehavior.ts]: typo ```recognize```

File: src/lib/accessibility/Behaviors/Menu/MenuItemBehavior.ts [hunk 13]
```diff
@@ -1,5 +1,14 @@
 import { Accessibility } from '../../interfaces'
 
+/**
+ * @description
+ * The MenuItemBehavior is done for structure of menu item which consist of root element and anchor inside the root element.
+ * The MenuItemBehavior adds :
+ * - role 'presentation' on the root element
+ * - role 'menuitem' on anchor element
+ * - 'aria-expanded' attribute on anchor element based on "submenuOpened" property
```
Comment (sophieH29) [inline src/lib/accessibility/Behaviors/Menu/MenuItemBehavior.ts]: I assume, add attribute ```to```, not ```on```. Please, correct me, if I am wrong

File: docs/src/components/DocsBehaviorRoot.tsx (hunk 5 above)
Comment (kolaps33) [reply docs/src/components/DocsBehaviorRoot.tsx]: I recognized that we need just Grid. Others imported components were leftovers. 
I took this part with the same styling from 'component' page. To have the same design. 
So if it necessary I can try use stardust grid.

File: src/lib/accessibility/Behaviors/Icon/IconBehavior.ts (hunk 6 above)
Comment (kolaps33) [reply src/lib/accessibility/Behaviors/Icon/IconBehavior.ts]: fixed :)

File: src/lib/accessibility/Behaviors/Button/ToggleButtonBehavior.ts (hunk 7 above)
Comment (kolaps33) [reply src/lib/accessibility/Behaviors/Button/ToggleButtonBehavior.ts]: fixed :)

File: src/lib/accessibility/Behaviors/Image/ImageBehavior.ts (hunk 8 above)
Comment (kolaps33) [reply src/lib/accessibility/Behaviors/Image/ImageBehavior.ts]: fixed

File: src/lib/accessibility/Behaviors/List/ListBehavior.ts (hunk 10 above)
Comment (kolaps33) [reply src/lib/accessibility/Behaviors/List/ListBehavior.ts]: fixed :)

File: src/lib/accessibility/Behaviors/List/BasicListItemBehavior.ts (hunk 9 above)
Comment (kolaps33) [reply src/lib/accessibility/Behaviors/List/BasicListItemBehavior.ts]: fixed :)

File: src/lib/accessibility/Behaviors/List/ListItemBehavior.ts (hunk 11 above)
Comment (kolaps33) [reply src/lib/accessibility/Behaviors/List/ListItemBehavior.ts]: fixed :)

File: src/lib/accessibility/Behaviors/List/SelectableListItemBehavior.ts (hunk 12 above)
Comment (kolaps33) [reply src/lib/accessibility/Behaviors/List/SelectableListItemBehavior.ts]: fixed :)

File: src/lib/accessibility/Behaviors/Menu/MenuItemBehavior.ts (hunk 13 above)
Comment (kolaps33) [reply src/lib/accessibility/Behaviors/Menu/MenuItemBehavior.ts]: fixed :)

File: docs/src/routes.tsx (hunk 4 above)
Comment (kolaps33) [reply docs/src/routes.tsx]: I need to help with this how to solve it..

File: build/gulp/plugins/gulp-component-menu-behaviors.ts [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+
+      let description
+      const fileContent = fs.readFileSync(file.path).toString()
+      const blockComments = extract(fileContent).filter(comment => comment.type === 'BlockComment') // filtering only block comments
+
+      // getting object that describes '@description' part of the comment's text
+      if (!_.isEmpty(blockComments)) {
+        const commentTokens = doctrine.parse(blockComments[0].raw, { unwrap: true }).tags
+        const descriptionToken = commentTokens.find(token => token.title === 'description')
+        description = descriptionToken.description
```
Comment (miroslavstastny) [inline build/gulp/plugins/gulp-component-menu-behaviors.ts]: Check `descriptionToken` is defined. Now if there is a block comment without `@description` build crashes.

File: build/gulp/tasks/docs.ts [hunk 15]
```diff
@@ -78,7 +91,12 @@ task('build:docs:example-menu', () =>
 
 task(
   'build:docs:json',
-  parallel('build:docs:docgen', 'build:docs:component-menu', 'build:docs:example-menu'),
+  parallel(
```
Comment (miroslavstastny) [inline build/gulp/tasks/docs.ts:96]: Watch (`yarn start`) does not reflect changes if Behavior description changes.

File: build/gulp/tasks/docs.ts [hunk 16]
```diff
@@ -55,6 +61,7 @@ task(
 // ----------------------------------------
 
 const componentsSrc = [`${config.paths.src()}/components/*/[A-Z]*.tsx`]
+const behaviorSrc = [`${config.paths.src()}/lib/accessibility/Behaviors/*/[A-Z]*.ts`]
```
Comment (miroslavstastny) [inline build/gulp/tasks/docs.ts:65]: What about `DefaultBehavior.ts`?

Comment (kolaps33) [reply build/gulp/tasks/docs.ts:65]: this is not real behavior which would need documentation, so it doesn't need be included

File: build/gulp/tasks/docs.ts (hunk 15 above)
Comment (kolaps33) [reply build/gulp/tasks/docs.ts:96]: good point :)
I was trying to implement that watch will rebuild either behaviors file, but was not able to find how to do it.
@miroslavstastny or @kuzhelov pleae if you could help me with it?

**Comment by miroslavstastny** [comment]
Also add `docs/src/componentMenuBehaviors.json` to `.gitignore`

File: build/gulp/tasks/docs.ts (hunk 15 above)
Comment (miroslavstastny) [reply build/gulp/tasks/docs.ts:96]: I added watch for behaviours. Watch currently does not work on Windows (not caused by this PR), I will address that in #172 once this PR is merged.

**Review by kuzhelov — APPROVED** [review]

File: docs/src/components/DocsBehaviorRoot.tsx [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+    children: PropTypes.node,
+    match: PropTypes.shape({
+      params: PropTypes.shape({
+        name: PropTypes.string.isRequired,
+      }),
+    }),
+  }
+
+  getNameFromFileName(fileName: string) {
+    const divided = _.startCase(fileName.replace('ts', ''))
```
Comment (kuzhelov) [inline docs/src/components/DocsBehaviorRoot.tsx]: it is better to introduce more descriptive name for this variable (like `baseName`)

File: docs/src/components/Sidebar/Sidebar.tsx [hunk 18]
```diff
@@ -12,10 +12,10 @@ import { getComponentPathname, typeOrder, repoURL } from 'docs/src/utils'
 
 const pkg = require('../../../../package.json')
 const componentMenu = require('docs/src/componentMenu')
+const behaviorMenuItems = require('docs/src/componentMenuBehaviors')
```
Comment (kuzhelov) [inline docs/src/components/Sidebar/Sidebar.tsx]: I would rather suggest to use `behaviorMenu` as a file name (and `behaviorMenu` as a variable name):

```js
const behaviorMenu = require('docs/src/behaviorMenu')
```

This will result in consistent pattern with the line above: 

```js
const componentMenu = require('docs/src/componentMenu')
```

**Comment by kuzhelov** [comment]
Couple of minor comments - please, take a look. 

Also, please, consider to review changes that were reported by Screener.

Thank you!

**Review by miroslavstastny — APPROVED** [review]
